iceoryx_examples/waitset/ice_waitset_publisher.cpp

```diff
@@ -49,7 +51,7 @@ void sending()
 
 int main()
 {
-    signal(SIGINT, sigHandler);
+    auto signalGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
```

Comment:
please register the signal handler either in `sending` or here. I suggest to do it in `main`